hibernate-core/src/main/java/org/hibernate/cfg/MultiTenancySettings.java

```diff
@@ -42,4 +42,17 @@ public interface MultiTenancySettings {
 	 * This setting configures the name of the DataSource to use for this access
 	 */
 	String TENANT_IDENTIFIER_TO_USE_FOR_ANY_KEY = "hibernate.multi_tenant.datasource.identifier_for_any";
+
+	/**
+	 * Specifies that {@link java.sql.Connection#setSchema(String)}}
+	 * should be called with the current tenant id, or with the schema name
+	 * returned by {@link CurrentTenantIdentifierResolver#schemaName}.
+	 *
+	 * @settingDefault false
+	 *
+	 * @see CurrentTenantIdentifierResolver#schemaName
+	 *
+	 * @since 7.1
+	 */
+	String SET_TENANT_SCHEMA = "hibernate.multi_tenant.set_schema";
```

Comment:
I don't love the naming here at all; unfortunately the existing properties have irregular naming.

Reply:
I think that was an unfortunate reality of a singular AvailableSettings with a zagillabillion settings.  Related things weren't always "geographically" close to one another and naming tended to get disjointed.  It was the main reason I started breaking them out.

But we can always "shadow deprecate" them - aka create new ones that replace the old while still recognizing the old.  See DeprecationLogger#deprecatedSetting(java.lang.String, java.lang.String)

Reply:
> But we can always "shadow deprecate" them - aka create new ones that replace the old while still recognizing the old. 

Yeah, that would be reasonable, I think.